receiver/syslogreceiver/syslog.go

```diff
@@ -0,0 +1,91 @@
+// Copyright 2021 OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package syslogreceiver
+
+import (
+	"github.com/open-telemetry/opentelemetry-log-collection/operator"
+	"github.com/open-telemetry/opentelemetry-log-collection/operator/builtin/input/syslog"
+	"github.com/open-telemetry/opentelemetry-log-collection/operator/builtin/input/tcp"
+	"github.com/open-telemetry/opentelemetry-log-collection/operator/builtin/input/udp"
+	syslogparser "github.com/open-telemetry/opentelemetry-log-collection/operator/builtin/parser/syslog"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/configmodels"
+	"gopkg.in/yaml.v2"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/stanza"
+)
+
+const typeStr = "syslog"
+
+// NewFactory creates a factory for filelog receiver
+func NewFactory() component.ReceiverFactory {
+	return stanza.NewFactory(ReceiverType{})
+}
+
+// ReceiverType implements stanza.LogReceiverType
+// to create a syslog receiver
+type ReceiverType struct{}
+
+// Type is the receiver type
+func (f ReceiverType) Type() configmodels.Type {
+	return configmodels.Type(typeStr)
+}
+
+// CreateDefaultConfig creates a config with type and version
+func (f ReceiverType) CreateDefaultConfig() configmodels.Receiver {
+	return &SysLogConfig{
+		BaseConfig: stanza.BaseConfig{
+			ReceiverSettings: configmodels.ReceiverSettings{
+				TypeVal: configmodels.Type(typeStr),
+				NameVal: typeStr,
+			},
+			Operators: stanza.OperatorConfigs{},
+		},
+		Input: stanza.InputConfig{},
+	}
+}
+
+// BaseConfig gets the base config from config, for now
+func (f ReceiverType) BaseConfig(cfg configmodels.Receiver) stanza.BaseConfig {
+	return cfg.(*SysLogConfig).BaseConfig
+}
+
+// SysLogConfig defines configuration for the syslog receiver
```

Comment:
The standard approach with the collector is to have a config.go file with this in it.

Please use the standard TLS config functionality from the collector. open telemetry-collector/config/configtls

Reply:
cool, got it, create an issue on openTelemetry log collection
https://github.com/open-telemetry/opentelemetry-log-collection/issues/59